Earn: Add navigation tabs

edanzer opened the pull request:
**Should be merged after https://github.com/Automattic/wp-calypso/pull/81697**

## Proposed Changes

- Adds the ability to add navigation tabs
- Puts all current content under the Tools tab
- Prepares the way to start adding other tabs per figma designs

**Before:**
<img width="1067" alt="no-tabs" src="https://github.com/Automattic/wp-calypso/assets/21228350/9134b89a-3fde-4bd7-873e-9626f5e30d04">

**After:**
<img width="1061" alt="earn-tabs" src="https://github.com/Automattic/wp-calypso/assets/21228350/f37d2188-cb5a-44c7-984d-a4874ef65c59">

**From Figma designs - what we're building toward**
<img width="694" alt="earn-design" src="https://github.com/Automattic/wp-calypso/assets/21228350/9e50e285-d4e0-4801-8674-e94c660aa1a7">


## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1) Checkout this branch and go to http://calypso.localhost:3000/earn/YOURDOMAIN
2) Confirm Tools tab shows as in screenshot above.
3) Navigate to some other Earn page (payments, ads, etc), click Tools tab, and confirm you are brought back to Earn home.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

edanzer committed:
Add tabs for Earn navigation

lezama commented:
Tests well on desktop ✅ 
On mobile I get this:

<img width="688" alt="Screenshot 2023-09-14 at 09 57 12" src="https://github.com/Automattic/wp-calypso/assets/104869/cefcb8e7-40cf-4545-a998-989452c9941e">

not sure what's the plan there.

edanzer committed:
Fix mobile styling

edanzer commented:
> Tests well on desktop ✅ On mobile I get this:
> ... not sure what's the plan there.

Good catch @lezama. To confirm, this is what this component looks like normally on mobile (this is from Earn > Ads, which uses the same component). 
<img width="348" alt="ad-tabs-mobile" src="https://github.com/Automattic/wp-calypso/assets/21228350/97dd3e32-a893-4af6-8295-4d1aceb8fd39">

I was overwriting the link color in this case, and I've ensure that doesn't apply on mobile. Now looks as follows. It is still a bit odd because we only have one tab. It'll look more normal when I add the second tab for Settings. 

<img width="298" alt="earn-tabs-mobile" src="https://github.com/Automattic/wp-calypso/assets/21228350/5e904145-ab05-42ea-9925-0aceceedefd3">

edanzer committed:
Remove some style overrides

edanzer commented:
There are some updated designs that don't require over-riding the SectionNav styles. I've updated screenshots in the PR description, and removed most style overrides to match.

lezama commented on client/my-sites/earn/style.scss:
```diff
@@ -50,3 +51,11 @@ See https://dequeuniversity.com/rules/axe/4.6/link-in-text-block?application=Axe
 	}
 }
+
+// Override some SectionNav styles on Earn page
```
do we really need them?

edanzer replied:
I suppose not. :) Removed.

edanzer committed:
Remove extra styles

edanzer committed:
Remove unneeded css import

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9069618

Thank you @edanzer for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9069618).
